src/memfd.rs

```diff
@@ -20,21 +20,59 @@ pub struct MemfdSettings {
 
 /// Create a memfd with the given settings.
 pub fn memfd_create(settings: MemfdSettings) -> Result<fs::File> {
-    let MemfdSettings { name, size } = settings;
+    #[cfg(not(target_os = "macos"))]
+    {
+        use std::ffi::CString;
+        let MemfdSettings { name, size } = settings;
 
-    let c_name = CString::new(name.clone()).context(error::NulZeroSnafu)?;
+        let c_name = CString::new(name.clone()).context(error::NulZeroSnafu)?;
+        // TODO: add sealing support. This can prevent some permission issues,
+        // such as a user without write permissions writing to the ringbuf.
+        let flags = memfd::MemFdCreateFlag::MFD_CLOEXEC;
 
-    let flags = memfd::MemFdCreateFlag::MFD_CLOEXEC;
+        let owned_fd = memfd::memfd_create(&c_name, flags)
+            .context(error::MemFdSnafu { fd_name: name })?;
 
-    let owned_fd = memfd::memfd_create(&c_name, flags)
-        .context(error::MemFdSnafu { fd_name: name })?;
+        let raw_fd = owned_fd.into_raw_fd();
 
-    let raw_fd = owned_fd.into_raw_fd();
+        let file = unsafe { fs::File::from_raw_fd(raw_fd) };
+        file.set_len(size).context(error::IoSnafu)?;
 
-    let file = unsafe { fs::File::from_raw_fd(raw_fd) };
-    file.set_len(size).context(error::IoSnafu)?;
+        Ok(file)
+    }
+    #[cfg(target_os = "macos")]
+    {
+        let MemfdSettings { name, size } = settings;
+
+        let base_path = String::from("/tmp/shm/");
```

Comment:
There seems to be no cleanup mechanism after normal files are created, which may result in more and more files being created.

Reply:
The file path doesn't seem to be configurable either.

Reply:
The new commit solved that.